packages/commerce-sdk-react/src/hooks/useDNT.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {useState} from 'react'
+import useAuthContext from './useAuthContext'
+import {getDefaultCookieAttributes} from '../utils'
+import Cookies from 'js-cookie'
+
+interface useDntReturn {
+    dntNotSet: boolean
+    updateDNT: (preference: boolean | null) => Promise<void>
+}
+
+/**
+ * Hook that updates the DNT preference and refreshes access token.
+ *
+ * @group Helpers
+ * @category Shopper Authentication
+ *
+ */
+const useDNT = (): useDntReturn => {
+    const auth = useAuthContext()
+
+    const dwDntValue = Cookies.get('dw_dnt')
+    const dntCookieIsDefined = dwDntValue !== '1' && dwDntValue !== '0'
+    const [dntNotSet, setDntNotSet] = useState(dntCookieIsDefined)
+
+    const updateDNT = async (preference: boolean | null) => {
+        let dntCookieVal = String(Number(preference))
+        // Use defaultDNT if defined. If not, use SLAS default DNT
+        if (preference === null)
+            dntCookieVal = auth.defaultDnt ? String(Number(auth.defaultDnt)) : '0'
+
+        // Set the cookie once to include dnt in the access token and then again to set the expiry time
+        Cookies.set('dw_dnt', dntCookieVal, {
+            ...getDefaultCookieAttributes()
+        })
+        setDntNotSet(false)
+        await auth.refreshAccessToken()
+
+        if (preference !== null) {
+            const daysUntilExpires = Number(auth.get('refresh_token_expires_in')) / 86400
+            Cookies.set('dw_dnt', dntCookieVal, {
+                ...getDefaultCookieAttributes(),
+                expires: daysUntilExpires
+            })
+        }
+    }
```

Comment:
If dw_dnt cookie is the SoT, we should call refresh token after the cookie has been settled. Why do we call it before cookie is set?

Reply:
The cookie is set twice because:
- First time: the refresh token flow needs to know of an updated `dw_dnt` value to put in as a claim, so cookie needs to be set before access token is refreshed. The part where `dw_dnt` is read in the refresh token flow is in Yuna's PR, but I'll add that here for clarity!
- Second time: The expiry time of the `dw_dnt` cookie needs to be set based on the refresh token expiry, only if a preference was given. If a preference were not given, then the cookie should expire on browser close.